### ISSUE TITLE
Failure with updated hyperxmp package

### ISSUE THREAD

**Issue by ppessl** [Cryptosaurus/iacrtrans#24]
Latex fails to render the template when updating the hyperxmp package to a recent version (>= 5.0). This is because the `\hyxmp@today` macro got removed, as used in the snippet below.

https://github.com/Cryptosaurus/iacrtrans/blob/b2c735585f3479750ff23acf4055df2990356b02/iacrtrans.cls#L451

A quick and very dirty fix might be to catch this with something like:
`\@ifundefined{hyxmp@today@xmp}{\hyxmp@today}{\hyxmp@today@xmp}`

I messaged the package author, Scott Pakin, on this topic, he replied with the following:
> \hyxmp@today is likely gone for good.
> 
> The reason iacrtrans.cls is accessing that package-internal macro is to  dynamically modify hyperxmp to include a document DOI.  However, the latest version of hyperxmp supports a pdfdoi key that iacrtrans can use instead.  The only difference is that hyperxmp expresses DOIs using the PRISM Basic Metadata Schema while iacrtrans's hack uses Dublin Core.
(The former supports DOIs, ISBNs, ISSNs, etc. in the same document while the latter allows at most one such identifier.)  Hopefully, the IACR folks don't care precisely how the DOI appears in the XMP metadata as long as it's present.
> 
> It would be possible for a class file to check the hyperxmp version and use either pdfdoi for new hyperxmp versions or the current rewriting hack for older ones, but I wouldn't recommend that approach on the grounds of it being awkward to maintain.

**Comment by junwei-wang** [comment]
Same issue to me

**Comment by Cryptosaurus** [comment]
Thanks @ppessl for the detailed bug report! (And sorry for the delay to address it)

I have updated to class to use pdfdoi when available, and to just skip the DOI in XMP metadata otherwise: 061fa4e85c4fdd8623a4da8e61acc80b5a3a9401
This should fix the issue and minimize the maintenance work in the future.

**Comment by junwei-wang** [comment]
The issue is still there:

```
ERROR: Undefined control sequence.

--- TeX said ---
<argument> \hyxmp@detect@langs 
                               
l.407 }
```

**Comment by Cryptosaurus** [comment]
Apparently, that was a different issue with hyperxmp 5.x (related to the order in which packages are loaded). Could check that it is fixed now?
If there is still an issue, please give a small example of a file that does not compile, and include the compilation log.

**Comment by junwei-wang** [comment]
I use MacTeX with latest packages from Ctan


```tex
\documentclass[final,journal=tches]{iacrtrans}

\title[short title]%
{Title}

\author{Some Author}

\institute{
  Some Company
}

\begin{document}
\maketitle

Some text


\end{document}
```

Log

```
$ pdflatex test.tex
This is pdfTeX, Version 3.14159265-2.6-1.40.21 (TeX Live 2020) (preloaded format=pdflatex)
 restricted \write18 enabled.
entering extended mode
(./test.tex
LaTeX2e <2020-02-02> patch level 5
L3 programming layer <2020-08-07> (./iacrtrans.cls

 *** LaTeX class for IACR Transactions v0.92 ***

Document Class: iacrtrans 2020/08/24
(/usr/local/texlive/2020/texmf-dist/tex/latex/xkeyval/xkeyval.sty
(/usr/local/texlive/2020/texmf-dist/tex/generic/xkeyval/xkeyval.tex
(/usr/local/texlive/2020/texmf-dist/tex/generic/xkeyval/xkvutils.tex
(/usr/local/texlive/2020/texmf-dist/tex/generic/xkeyval/keyval.tex))))
(/usr/local/texlive/2020/texmf-dist/tex/latex/base/article.cls
Document Class: article 2019/12/20 v1.4l Standard LaTeX document class
(/usr/local/texlive/2020/texmf-dist/tex/latex/base/size10.clo))
(/usr/local/texlive/2020/texmf-dist/tex/latex/geometry/geometry.sty
(/usr/local/texlive/2020/texmf-dist/tex/generic/iftex/ifvtex.sty
(/usr/local/texlive/2020/texmf-dist/tex/generic/iftex/iftex.sty)))
(/usr/local/texlive/2020/texmf-dist/tex/latex/tools/afterpage.sty)
(/usr/local/texlive/2020/texmf-dist/tex/latex/sectsty/sectsty.sty

LaTeX Warning: Command \underbar  has changed.
               Check if current package is valid.


LaTeX Warning: Command \underline  has changed.
               Check if current package is valid.

) (/usr/local/texlive/2020/texmf-dist/tex/latex/fancyhdr/fancyhdr.sty)
(/usr/local/texlive/2020/texmf-dist/tex/latex/graphics/graphicx.sty
(/usr/local/texlive/2020/texmf-dist/tex/latex/graphics/graphics.sty
(/usr/local/texlive/2020/texmf-dist/tex/latex/graphics/trig.sty)
(/usr/local/texlive/2020/texmf-dist/tex/latex/graphics-cfg/graphics.cfg)
(/usr/local/texlive/2020/texmf-dist/tex/latex/graphics-def/pdftex.def)))
(/usr/local/texlive/2020/texmf-dist/tex/latex/lastpage/lastpage.sty)
(/usr/local/texlive/2020/texmf-dist/tex/latex/xcolor/xcolor.sty
(/usr/local/texlive/2020/texmf-dist/tex/latex/graphics-cfg/color.cfg))
(/usr/local/texlive/2020/texmf-dist/tex/latex/etoolbox/etoolbox.sty)
(/usr/local/texlive/2020/texmf-dist/tex/latex/amsmath/amsmath.sty
For additional information on amsmath, use the `?' option.
(/usr/local/texlive/2020/texmf-dist/tex/latex/amsmath/amstext.sty
(/usr/local/texlive/2020/texmf-dist/tex/latex/amsmath/amsgen.sty))
(/usr/local/texlive/2020/texmf-dist/tex/latex/amsmath/amsbsy.sty)
(/usr/local/texlive/2020/texmf-dist/tex/latex/amsmath/amsopn.sty))
(/usr/local/texlive/2020/texmf-dist/tex/latex/amsfonts/amssymb.sty
(/usr/local/texlive/2020/texmf-dist/tex/latex/amsfonts/amsfonts.sty))
(/usr/local/texlive/2020/texmf-dist/tex/latex/amscls/amsthm.sty)
(/usr/local/texlive/2020/texmf-dist/tex/latex/mathtools/mathtools.sty
(/usr/local/texlive/2020/texmf-dist/tex/latex/tools/calc.sty)
(/usr/local/texlive/2020/texmf-dist/tex/latex/mathtools/mhsetup.sty))
(/usr/local/texlive/2020/texmf-dist/tex/latex/float/float.sty)
(/usr/local/texlive/2020/texmf-dist/tex/latex/microtype/microtype.sty
(/usr/local/texlive/2020/texmf-dist/tex/latex/microtype/microtype-pdftex.def)
(/usr/local/texlive/2020/texmf-dist/tex/latex/microtype/microtype.cfg))
(/usr/local/texlive/2020/texmf-dist/tex/latex/base/fontenc.sty)
(/usr/local/texlive/2020/texmf-dist/tex/latex/lm/lmodern.sty))
(/usr/local/texlive/2020/texmf-dist/tex/latex/hyperref/hyperref.sty
(/usr/local/texlive/2020/texmf-dist/tex/generic/ltxcmds/ltxcmds.sty)
(/usr/local/texlive/2020/texmf-dist/tex/generic/pdftexcmds/pdftexcmds.sty
(/usr/local/texlive/2020/texmf-dist/tex/generic/infwarerr/infwarerr.sty))
(/usr/local/texlive/2020/texmf-dist/tex/generic/kvsetkeys/kvsetkeys.sty)
(/usr/local/texlive/2020/texmf-dist/tex/generic/kvdefinekeys/kvdefinekeys.sty)
(/usr/local/texlive/2020/texmf-dist/tex/generic/pdfescape/pdfescape.sty)
(/usr/local/texlive/2020/texmf-dist/tex/latex/hycolor/hycolor.sty)
(/usr/local/texlive/2020/texmf-dist/tex/latex/letltxmacro/letltxmacro.sty)
(/usr/local/texlive/2020/texmf-dist/tex/latex/auxhook/auxhook.sty)
(/usr/local/texlive/2020/texmf-dist/tex/latex/kvoptions/kvoptions.sty)
(/usr/local/texlive/2020/texmf-dist/tex/latex/hyperref/pd1enc.def)
(/usr/local/texlive/2020/texmf-dist/tex/generic/intcalc/intcalc.sty)
(/usr/local/texlive/2020/texmf-dist/tex/generic/etexcmds/etexcmds.sty)
(/usr/local/texlive/2020/texmf-dist/tex/latex/url/url.sty)
(/usr/local/texlive/2020/texmf-dist/tex/generic/bitset/bitset.sty
(/usr/local/texlive/2020/texmf-dist/tex/generic/bigintcalc/bigintcalc.sty))
(/usr/local/texlive/2020/texmf-dist/tex/generic/atbegshi/atbegshi.sty))
(/usr/local/texlive/2020/texmf-dist/tex/latex/hyperref/hpdftex.def
(/usr/local/texlive/2020/texmf-dist/tex/latex/atveryend/atveryend.sty)
(/usr/local/texlive/2020/texmf-dist/tex/latex/rerunfilecheck/rerunfilecheck.sty

(/usr/local/texlive/2020/texmf-dist/tex/generic/uniquecounter/uniquecounter.sty
))) (/usr/local/texlive/2020/texmf-dist/tex/latex/hyperxmp/hyperxmp.sty
(/usr/local/texlive/2020/texmf-dist/tex/generic/stringenc/stringenc.sty)
(/usr/local/texlive/2020/texmf-dist/tex/latex/ifmtarg/ifmtarg.sty)
(/usr/local/texlive/2020/texmf-dist/tex/latex/base/ifthen.sty)
(/usr/local/texlive/2020/texmf-dist/tex/latex/oberdiek/ifdraft.sty)
! Undefined control sequence.
<argument> \hyxmp@detect@langs

l.407 }

?
```

**Comment by Cryptosaurus** [comment]
Just to be sure, are you using the latest iacrtrans.cls?  (I have updated it a few hours ago, but I didn't change the version number)

I don't have a Mac, but MacTex is based on TeX Live 2020, and your example file compiles fine with the TeX Live from Ubuntu Groovy...

**Comment by junwei-wang** [comment]
Yes, I did. As you can see from the log

```
 *** LaTeX class for IACR Transactions v0.92 ***

Document Class: iacrtrans 2020/08/24
```

I use MacTeX and all packages have been updated to the latest versions by using

```sh
tlmgr update --self --all
```

**Comment by Cryptosaurus** [comment]
I've just tried your example with a fresh install of MacTex and it works fine (before or after `tlmgr update --self --all`).
Could you retry with a fresh download of iacrtrans.cls (version 6e53867841a955805e214545beb30a62dfe1461d had an issue with the same type of error message, but it's been fixed in 44ffcb635889d4d78bae32a9f436380fa6f83caf)?

**Comment by junwei-wang** [comment]
I re-tried compilation after cleaning temporary files. Everything works fine now.

Sorry about the previous messages.
